DataFormats/TrackerRecHit2D/interface/SiPixelRecHitQuality.h

```diff
@@ -123,6 +123,9 @@ class SiPixelRecHitQuality {
     inline void setProbabilityQ(float prob, QualWordType& qualWord) const {
       if (prob > 1.0f && prob <= 1.0f + std::numeric_limits<float>::epsilon()) {
         prob = 1;
+      } else if (prob == -1) {
```

Comment:
```suggestion
      } else if (prob == -1. 0f) {
```

shouldn't it be compared against floating point representation?

Reply:
Done in [bd6dab7](https://github.com/cms-sw/cmssw/pull/39776/commits/bd6dab721918035528b1471ff9db5760384e1d52).